app/js/app/directives/graph_sketcher/GraphPreview.js

```diff
@@ -11,392 +11,51 @@ define(function(require) {
-                        p.textSize(14);
-                        p.text(symbol.text, symbol.x - 4, symbol.y + 20);
-                        
-                        p.pop();
-                    }
-
-                    function drawSymbols(symbols, color) {  
-                        for (var i = 0; i < symbols.length; i++) {
-                            drawSymbol(symbols[i], color);
-                        }
-                    }
-
-                    function drawVerticalDotLine(x, begin, end) {
-                        if (x < 0 || x > canvasWidth) {
-                            return;
-                        }
-
-                        if (begin > end) {
-                            var tmp = begin;
-                            begin = end;
-                            end = tmp;
-                        }
-
-                        p.push();
-                        p.stroke(DOT_LINE_COLOR);
-                        p.strokeWeight(CURVE_STRKWEIGHT);
-
-                        var step = DOT_LINE_STEP;
-                        var toDraw = true;
-                        var y = begin;
-                        while (y + step < end) {
-                            if (toDraw) {
-                                p.line(x, y, x, y+step);
-                            }
-                            y += step;
-                            toDraw = !toDraw;
-                        }
-                        if (toDraw) {
-                            p.line(x, y, x, end);
-                        }
-
-                        p.pop();
-                    }
-
-                    function drawHorizontalDotLine(y, begin, end) {
-                        if (y < 0 || y > canvasHeight) {
-                            return;
-                        }
-
-                        if (begin > end) {
-                            var tmp = begin;
-                            begin = end;
-                            end = tmp;
-                        }
-
-                        p.push();
-                        p.stroke(DOT_LINE_COLOR);
-                        p.strokeWeight(CURVE_STRKWEIGHT);
-
-                        var step = DOT_LINE_STEP;
-                        var toDraw = true;
-                        var x = begin;
-                        while (x + step < end) {
-                            if (toDraw) {
-                                p.line(x, y, x+step, y);
-                            }
-                            x += step;
-                            toDraw = !toDraw;
-                        }
-                        if (toDraw) {
-                            p.line(x, y, end, y);
-                        }
-
-                        p.pop();
-                    }
-
-                    function clone(obj) {
-                        var json = JSON.stringify(obj);
-                        return JSON.parse(json);
-                    }
-                  
                     function decodeData(rawData) {
```

Comment:
`decodeData` is probably almost identical to the one used in GraphSketcher so is a good candidate for refactoring out in the future (possibly into GraphUtils).